src/solvers/smt2/smt2_solver.cpp

```diff
@@ -162,6 +178,13 @@ void smt2_solvert::command(const std::string &c)
         std::cout << e.pretty() << '\n'; // need to do an 'smt2_format'
       }
     }
+    else if(c == "echo")
+    {
+      if(next_token() != STRING_LITERAL)
+        std::cout << "expected string literal" << '\n';
+      else
+        std::cout << smt2_string_literal(buffer) << '\n';
```

Comment:
It seems errors and desired output are indistinguishable?

Reply:
Needs #3441